App/Dependencies/ExposureDetection/ImmuniTemporaryExposureKeyProvider.swift

```diff
@@ -49,18 +57,36 @@ class ImmuniTemporaryExposureKeyProvider: TemporaryExposureKeyProvider {
       .delete(chunks.flatMap { $0.localUrls })
   }
 
-  private func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
+  func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
     self.networkManager.getKeysIndex()
       .then { (keysIndex: KeysIndex) -> [Int] in
-        guard let latestKnownChunkIndex = latestKnownChunkIndex else {
-          return Array(keysIndex.oldest ... keysIndex.newest)
-        }
 
-        guard keysIndex.newest > latestKnownChunkIndex else {
+        let latestKnown = latestKnownChunkIndex ?? -1
+
+        guard keysIndex.newest > latestKnown else {
+          // no chunks to download
           return []
         }
 
-        return Array(keysIndex.oldest.bounded(min: latestKnownChunkIndex) ... keysIndex.newest)
+        let upperBound = keysIndex.newest
```

Comment:
I feel this logic can be made more readable by refactoring it to something like:
```swift
let firstKeyToDownload = max(latestKnown + 1, keysIndex.oldest)
return (firstKeyToDownload...keysIndex.newest).suffix(Self.keyDailyRateLimit)
```

The `suffix` makes more explicit that we're giving priority to the newest, and the logic for choosing the starting point is all in `firstKeyToDownload`